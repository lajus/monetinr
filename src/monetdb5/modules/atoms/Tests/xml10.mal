# produce a more complex structure
# such as 
#<books>
#  <author> john
#  <title> spring</title>
#  <title> summer</title>
#  </author>
#  <author> mary
#  <title> autum</title>
#  </author>
#</books>

# the intermediate tabular result produced by SQL
a:= bat.new(:oid,:str);
bat.insert(a,0@0,"john");
bat.insert(a,1@0,"john");
bat.insert(a,2@0,"marie");
t:= bat.new(:oid,:str);
bat.insert(t,0@0,"spring");
bat.insert(t,1@0,"summer");
bat.insert(t,2@0,"autum");

# SQL rendering request
#xmlelement('books',
#	xmlelement('author',a,
#			xmlagg(xmlelement('title',t))
#   ) )
# every nesting implies a groupby

ax:= xml.xml(a);
tx:= xml.xml(t);
te:= xml.element("title",tx);
g:= bat.new(:oid,:oid);
bat.insert(g,0@0,0@0);
bat.insert(g,1@0,0@0);
bat.insert(g,2@0,1@0);

k:= algebra.join(g,te);
io.print(k);
l:= algebra.sort(k);
io.print(l);

ag:= xml.agg(te,g);
io.print(ag);

gr:= bat.reverse(g);
j2:= algebra.join(gr,ax);
k2:=algebra.kunique(j2);
io.print(k2);
cc:= xml.concat(k2,ag);
io.print(cc);
io.print("author construction");
ae:= xml.element("author",cc);
io.print(ae);

io.print("book construction");
be:= xml.agg(ae);
io.print(be);
doc:= xml.element("books",be);
io.print(doc);
