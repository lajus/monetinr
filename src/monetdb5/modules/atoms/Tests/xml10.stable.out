stdout of test 'xml10` in directory 'monetdb5/modules/atoms` itself:


# 09:38:47 >  
# 09:38:47 >  Mtimeout -timeout 60  mserver5 "--config=/ufs/mk/monet5/Linux/etc/monetdb5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5//Linux/lib/MonetDB5:/ufs/mk/monet5//Linux/lib/MonetDB5/lib:/ufs/mk/monet5//Linux/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/mk/monet5//Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/sql_logs" --set "xquery_logdir=/ufs/mk/monet5//Linux/var/MonetDB5/xquery_logs" --set mapi_open=true --set xrpc_open=true --set mapi_port=37982 --set xrpc_port=42894 --set monet_prompt= --trace --dbname=mTests_src_modules_mal  xml10.mal
# 09:38:47 >  

# MonetDB server v5.1.0, based on kernel v1.19.0
# Serving database 'mTests_src_modules_mal'
# Compiled for x86_64-redhat-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-2007 CWI, all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://eir.ins.cwi.nl:34949/
# MonetDB/SQL module v2.39.0 loaded
# MonetDB/GIS module v0.19.0 loaded

function user.main():void;
# produce a more complex structure 
# such as  
#<books> 
#  <author> john 
#  <title> spring</title> 
#  <title> summer</title> 
#  </author> 
#  <author> mary 
#  <title> autum</title> 
#  </author> 
#</books> 
# the intermediate tabular result produced by SQL 
    a := bat.new(:oid,:str);
    bat.insert(a,0@0,"john");
    bat.insert(a,1@0,"john");
    bat.insert(a,2@0,"marie");
    t := bat.new(:oid,:str);
    bat.insert(t,0@0,"spring");
    bat.insert(t,1@0,"summer");
    bat.insert(t,2@0,"autum");
# SQL rendering request 
#xmlelement('books', 
#	xmlelement('author',a, 
#			xmlagg(xmlelement('title',t)) 
#   ) ) 
# every nesting implies a groupby 
    ax := xml.xml(a);
    tx := xml.xml(t);
    te := xml.element("title",tx);
    g := bat.new(:oid,:oid);
    bat.insert(g,0@0,0@0);
    bat.insert(g,1@0,0@0);
    bat.insert(g,2@0,1@0);
    k := algebra.join(g,te);
    io.print(k);
    l := algebra.sort(k);
    io.print(l);
    ag := xml.agg(te,g);
    io.print(ag);
    gr := bat.reverse(g);
    j2 := algebra.join(gr,ax);
    k2 := algebra.kunique(j2);
    io.print(k2);
    cc := xml.concat(k2,ag);
    io.print(cc);
    io.print("author construction");
    ae := xml.element("author",cc);
    io.print(ae);
    io.print("book construction");
    be := xml.agg(ae);
    io.print(be);
    doc := xml.element("books",be);
    io.print(doc);
end main;
#---------------------------------#
# h	t							  # name
# void	xml			  # type
#---------------------------------#
[ 0@0,	  <title>spring</title>	  ]
[ 1@0,	  <title>spring</title>	  ]
[ 2@0,	  <title>summer</title>	  ]
#---------------------------------#
# h	t																  # name
# void	xml			  # type
#---------------------------------#
[ 0@0,	  <title>spring</title>	  ]
[ 1@0,	  <title>spring</title>	  ]
[ 2@0,	  <title>summer</title>	  ]
#---------------------------------------------------------#
# h	t						  # name
# void	xml						  # type
#---------------------------------------------------------#
[ 0@0,	  <title>spring</title><title>summer</title>	  ]
[ 1@0,	  <title>autum</title>				  ]
#-----------------#
# h	t	  # name
# oid	xml	  # type
#-----------------#
[ 0@0,	  john	  ]
[ 1@0,	  marie	  ]
#---------------------------------------------------------#
# h	t						  # name
# oid	xml						  # type
#---------------------------------------------------------#
[ 0@0,	  john<title>spring</title><title>summer</title>  ]
[ 1@0,	  marie<title>autum</title>			  ]
[ "author construction" ]
#---------------------------------------------------------------------------------#
# h	t									  # name
# oid	xml									  # type
#---------------------------------------------------------------------------------#
[ 0@0,	  <author>john<title>spring</title><title>summer</title></author>	  ]
[ 1@0,	  <author>marie<title>autum</title></author>				  ]
[ "book construction" ]
[ <author>john<title>spring</title><title>summer</title></author><author>marie<title>autum</title></author> ]
[ <books><author>john<title>spring</title><title>summer</title></author><author>marie<title>autum</title></author></books> ]

# 23:09:29 >  
# 23:09:29 >  Done.
# 23:09:29 >  

